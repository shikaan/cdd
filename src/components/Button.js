import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const getColorFromContext = (context, theme) => {
  if (context === CONTEXT.ACCENT) {
    return `
      border: none;
      background-color: ${theme.color.brand400};
      color: ${theme.color.light}
    `;
  } else if (context === CONTEXT.OUTLINE_LIGHT) {
    return `
      border: 1px solid ${theme.color.light};
      background-color: transparent;
      color: ${theme.color.light};
    `;
  } else {
    return null;
  }
};

export const CONTEXT = {
  ACCENT: "accent",
  OUTLINE_LIGHT: "outline-light",
};

const StyledButton = styled.button(({theme, context}) => `
  font-family: ${theme.typography.primaryFont.fontFamily};
  font-size: ${theme.typography.baseFontSize.multiply(0.875)};
  font-weight: 700;
  padding: ${theme.typography.baseFontSize.multiply(0.5)} ${theme.typography.baseFontSize.multiply(1.5)};
  border-radius: 2px;

  ${getColorFromContext(context, theme)}  
`);

const Button = (props) => <StyledButton {...props} />;

Button.propTypes = {
  context: PropTypes.oneOf(Object.values(CONTEXT)),
  onClick: PropTypes.func
};

export default Button;
