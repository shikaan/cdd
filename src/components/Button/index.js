import React from "react";

export default class Button extends React.Component {
  render() {
    const {children, onClick} = this.props

    return <button onClick={onClick}>{children}</button>
  }
}
