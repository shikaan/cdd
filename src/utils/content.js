/**
 * @param {Array<{reference: string}>} microcopy
 * @param {string} key
 * @return {string}
 */
export const getMicrocopy = (microcopy, key) => {
  const element = microcopy?.find(i => i.reference === key) ?? {};

  return element.value ?? "";
};

/**
 * @param {Array<{reference: string}>} sections
 * @param {string} key
 * @return {string}
 */
export const getSection = (sections, key) => {
  return sections?.find(i => i.reference === key) ?? {reference: key};
};