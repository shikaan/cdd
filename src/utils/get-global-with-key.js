export const getGlobalWithKey = (key = "location") => {
  return typeof window !== "undefined" ? window : {[key]: {}};
};
