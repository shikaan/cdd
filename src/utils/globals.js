export const getGlobalWithKey = (key = "location") => {
  return typeof window !== "undefined" ? window : {[key]: {}};
};

export const getGlobal = () => { 
  // eslint-disable-next-line no-restricted-globals
  if (typeof self !== "undefined") { return self }
  if (typeof window !== "undefined") { return window } 
  if (typeof global !== "undefined") { return global } 
  throw new Error("unable to locate global object"); 
}; 
