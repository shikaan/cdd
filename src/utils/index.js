export {isLastIndex} from "./is-last-index";
