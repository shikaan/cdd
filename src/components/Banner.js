import React from "react";
import styled from "styled-components";
import Button, {CONTEXT} from "~components/Button";

const BannerWrapper = styled.div(({theme}) => `
  width: 100%;
  padding: ${theme.typography.baseFontSize};
  background-color: ${theme.color.brand300};
  display: flex;
`);

const BannerText = styled.span(({theme}) => `
  flex: 11;
  color: ${theme.color.light};
  line-height: 40px;
  vertical-align: middle;
`);

const BannerButtonWrapper = styled.div(({theme}) => `
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: center;
`);

export default ({children, ctaLabel, ctaAction, ...rest}) => {
  return (
    <BannerWrapper {...rest} >
      <BannerText>
        {children}
      </BannerText>
      {
        ctaLabel &&
        <BannerButtonWrapper>
          <Button context={CONTEXT.OUTLINE_LIGHT} onClick={ctaAction}>
            {ctaLabel}
          </Button>
        </BannerButtonWrapper>
      }
    </BannerWrapper>
  );
};
