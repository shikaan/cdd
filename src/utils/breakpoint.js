import {numberBreakpoint} from "~theme/variables";
import {getGlobalWithKey} from "~/utils/get-global-with-key";

export const getViewportWidth = () => getGlobalWithKey("innerWidth") ?? getGlobalWithKey("document").documentElement.clientWidth;

export const isMobile = () => {
  const viewportWidth = getViewportWidth();
  return viewportWidth < numberBreakpoint.sm;
};

export const isTablet = () => {
  const viewportWidth = getViewportWidth();
  return numberBreakpoint.sm <= viewportWidth && viewportWidth < numberBreakpoint.md;
};

export const isDesktop = () => {
  const viewportWidth = getViewportWidth();
  return numberBreakpoint.md <= viewportWidth;
};