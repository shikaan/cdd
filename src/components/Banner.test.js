import React from "react";
import {shallow} from "enzyme";
import "jest-styled-components";
import {mountWithTheme} from "/test/utils";

import themeMock from "~theme";
import Button from "~components/Button";

import Banner from "./Banner.js";

describe("Banner", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Banner/>);

    expect(wrapper).toMatchSnapshot();
  });

  it("calls ctaAction when button is clicked", () => {
    const ctaActionStub = jest.fn();
    const wrapper = mountWithTheme(<Banner ctaLabel={"label"} ctaAction={ctaActionStub}/>, themeMock);

    wrapper.find(Button).simulate("click");

    expect(ctaActionStub).toHaveBeenCalled();
  });

  it("renders cta only if label is provided", () => {
    const wrapperWithoutCTA = mountWithTheme(<Banner />, themeMock);
    const wrapperWithCTA = mountWithTheme(<Banner ctaLabel={"label"} />, themeMock);

    expect(wrapperWithoutCTA.find(Button)).toHaveLength(0);
    expect(wrapperWithCTA.find(Button)).not.toHaveLength(0);
  });
});
