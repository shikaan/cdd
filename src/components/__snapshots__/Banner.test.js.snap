// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders correctly 1`] = `
<styled.div>
  <styled.span />
</styled.div>
`;
